pkg/config/utils.go

```diff
@@ -434,7 +434,9 @@ func ReplaceContextTokens(context Context, pattern string) string {
 		strings.Replace(
 			strings.Replace(
 				strings.Replace(
-					strings.Replace(pattern,
+					strings.Replace(
+						strings.Replace(pattern,
+							"{base-component}", context.BaseComponent, 1),
```

Comment:
You can use the following to make this a bit cleaner...

```go
r := strings.NewReplacer(
	"{base-component}", context.BaseComponent,
	"{component}", "context.Component",
	...other find/replace pairs)

    
    // Replace all pairs.
    result := r.Replace(value)
```

Reply:
this is nice (easier to read), thanks @mcalhoun , updated